alibi/explainers/anchor_tabular.py

```diff
@@ -958,30 +979,64 @@ def add_names_to_exp(self, explanation: dict) -> None:
                 handled.add(feat_id)
             explanation['names'].append(fname)
 
+    @property
+    def predictor(self) -> Callable:
+        return self._ohe_predictor if self.ohe else self._predictor
+
+    @predictor.setter
+    def predictor(self, predictor: Callable) -> None:
+        # if input is one-hot encoded
+        if self.ohe:
+            # this predictor expects ordinal/labels encoded categorical variables
+            ord_predictor = lambda x: predictor(ord_to_ohe(x, self.cat_vars_ord)[0])  # noqa: E731
+            self._predictor = self._transform_predictor(ord_predictor) if predictor else None
+
+            # this predictor expects one-hot encoded categorical variable
+            self._ohe_predictor = self._transform_ohe_predictor(predictor) if predictor else None
+
+        else:
+            # set the predictor
+            self._predictor = self._transform_predictor(predictor) if predictor else None
```

Comment:
What's the reason for these `if predictor else None`? The `predictor` is typed as `Callable` so `None` should not be passed, we probably shouldn't silently set `self._predictor` etc. to `None` unless I'm missing something?

Reply:
`explainer.predictor = None` in the tests.

https://github.com/SeldonIO/alibi/blob/af1b3513f000daf4061eb20510ad539e70edeebc/alibi/saving.py#L101-L106

Reply:
Great, specifying the type hint for the setter to `Optional[Callable]` should be then good to go.